### DISCUSSION TITLE
Trouble getting tagged pages to show up on tag archive/tags list

### DISCUSSION THREAD

**Discussion by vortexing** [mmistakes/minimal-mistakes#2627] [Q&A] [answered]
I'm hoping this is just me being rusty with my Jekyll/liquid stuff, but I'm also wondering if it's just that we've really accumulated some "history" in this repo.  We set this up long ago and now that it's worked for so long, I find myself struggling again with liquid.  

We are wanting to create a page like this one:
https://mmistakes.github.io/minimal-mistakes/tags/
And start adding tags to our pages.  We'd also like to create a landing page for each of the tags themselves like this:
https://mmistakes.github.io/minimal-mistakes/tags/markup/

We're using Github pages, and our repo and my most recent attempt is here:
https://github.com/FredHutch/wiki/tree/pathways

I can get the above pages to render, but they don't have any tagged pages actually ON them.  We aren't using a `_posts` folder nor a `_pages` folder but rather multiple collections.  Is there an issue with getting a list of tagged pages in collections on the site to show up on the tags page because they aren't posts???  Am I doing something super simple wrong???

**Comment by mmistakes** [comment, marked as answer]
>We aren't using a _posts folder nor a _pages folder but rather multiple collections. 

This right here is the issue. Jekyll only supports "site wide" tags on documents in `_posts/`. You can add tags to pages and collections, but they won't be part of the `site.tags` object used to generate the tag indexes.

You can look through some old issues on [jekyll/jekyll](https://github.com/jekyll/jekyll/issues) where this issue was raised. There have been talks over the years to [harmonize posts, pages, collections,](https://github.com/jekyll/jekyll/issues/7296) so the behavior of `categories` and `tags` are consistent between... but to my knowledge there's be no movement on that front.

**Comment by vortexing** [comment]
oooof.  I was afraid of that.  Sigh.  So, can you imagine a way to add front matter metadata to pages in collections and then collect, over all collections, tagged pages and make links to them?  

Basically I want the structure of the collections to organize the pages, but then make little "cheat sheets" that are lists of pages to go to that have a specific tag (or something like a tag) on them.  Is there another approach that mimics this?

**Comment by mmistakes** [comment]
Since tags are only supported on posts, I never tried adding them to collections or pages.
No idea if these exist, but you could try a test page and see if it returns any data:

For pages:

```
{{ site.pages.tags | inspect }}
```

For a collection (replace `collection_name` with the name used to define the collection in `_config.yml`

```
{{ site.collection_name.tags | inspect }}
```

If these return document data, then you could modify my post tags template to loop over `site.pages.tags` or `site.collection_name.tags` instead of `site.tags`. You'll probably need to combine those objects too as you'll have a set of documents in the `pages` object and each of your collections.

You might want to search through the [Jekyll Talk forum](https://talk.jekyllrb.com/search?q=collection%20tag) as I know I've seen others wanting to do the same. Just don't think there was a solution until Jekyll core did something to make `site.tags` reflect **all** documents and not just posts.

**Comment by mmistakes** [comment]
Closing as this an upstream issue.

**Comment by vortexing** [comment]
Thanks for the tidbit there.  From what I can manage to wrangle, I can get the `site.<collection>.tags` to be listed on a page.  

I'm not sure (again, liquid/jekyll/ruby rookie), how to loop over all the collections, find any pages with a specific tag, and then show those page url's as if it was on your `posts-tag.html` template.  

We'll be manually creating tags and there won't be many.  So we can make a markdown for each of the tags to be a landing page for this.  On that markdown, I can't figure out how to do it, but I need to loop through all the collections on the site (`site.collections`?), find every page with the specific tag, then make it pop up as if it's on your `posts-tag.html` template. Then on each of the markdowns that is a landing page for a tag list, I can just hard code the tag that I want to show on that page.  That would be a totally reasonable solution.... But newb.  (all the tidbits on the Jekyll forum seem to be either not simple enough for me to understand or too simple b/c there's only one collection).

**Comment by mmistakes** [comment]
Yeah I dunno. Jekyll Talk or Stackoverflow are your best bet at this point. The fact that I see this use case come up so often tells me no one has a good solution for it so you might be out of luck.

I also just went with posts and used `categories` as a way of grouping them. It's not as semantic as having a dedicated collection, but nothing stopping you from organizing them like:

```
├── _posts
|  ├── fake-collection-1
|       └── YYYY-MM-DD-document-1.md
|  └── fake-collection-2
|       └── YYYY-MM-DD-document-2.md
|       └── YYYY-MM-DD-document-3.md
```
